network-store-client/src/main/java/com/powsybl/network/store/client/CollectionBuffer.java

```diff
@@ -45,11 +71,22 @@ void create(Resource<T> resource) {
     }
 
     void update(Resource<T> resource) {
+        update(resource, null);
+    }
+
+    void update(Resource<T> resource, AttributeFilter attributeFilter) {
         // do not update the resource if a creation resource is already in the buffer
         // (so we don't need to generate an update as the resource has not yet been created
         // on server side and is still on client buffer)
         if (!createResources.containsKey(resource.getId())) {
-            updateResources.put(resource.getId(), resource);
+            ResourceAndFilter<T> resourceAndFilter = updateResources.get(resource.getId());
+            if (resourceAndFilter == null) {
+                updateResources.put(resource.getId(), new ResourceAndFilter<>(resource, attributeFilter));
+            } else {
+                if (resourceAndFilter.getAttributeFilter() != null && attributeFilter == null) {
```

Comment:
Add a comment here to say that we may need to handle more "merges" of attribute filters when we create new types ?

Reply:
comment added